### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by tarun-29**
This PR includes following changes
1. Introduce custom hooks to separate logic and JSX
2. Code refactoring using ESLint and Prettier
3. Add prettier package

### CODE REVIEW

**Comment by lucifercr07** [comment]
@tarun-29 please resolve conflicts.

**Comment by tarun-29** [comment]
> @tarun-29 please resolve conflicts.

This is resolved

**Comment by tarun-29** [comment]
@lucifercr07 i have removed file which was committed by mistake the build should go through now

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "lucide-react": "^0.446.0",
     "next": "14.2.13",
     "package.json": "^2.0.1",
+    "prettier": "^3.3.3",
```
Comment (lucifercr07) [inline package.json:21]: Can you add this as part of scripts?

Comment (lucifercr07) [reply package.json:21]: Also if you can add this as a Github workflow would be helpful, may be as separate PR this can be taken up.

**Review by lucifercr07 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (lucifercr07) [reply package.json:21]: @tarun-29 please feel free to create new issue and fix this.

**Pull request merged**